v3-sdk/routing/src/libs/providers.ts

```diff
@@ -0,0 +1,122 @@
+import { ethers, providers, BigNumber } from 'ethers'
+import { Environment, CurrentConfig } from '../config'
+import { BaseProvider } from '@ethersproject/providers'
+
+// Single copies of provider and wallet
+const mainnetProvider = new ethers.providers.JsonRpcProvider(
+  CurrentConfig.rpc.mainnet
+)
+const wallet = createWallet()
+
+const browserExtensionProvider = createBrowserExtensionProvider()
+let walletExtensionAddress: string | null = null
+
+// Interfaces
+
+export enum TransactionState {
+  Failed = 'Failed',
+  New = 'New',
+  Rejected = 'Rejected',
+  Sending = 'Sending',
+  Sent = 'Sent',
+}
+
+// Provider and Wallet Functions
+
+export function getMainnetProvider(): BaseProvider {
+  return mainnetProvider
+}
+
+export function getProvider(): providers.Provider | null {
+  return CurrentConfig.env === Environment.WALLET_EXTENSION
+    ? browserExtensionProvider
+    : wallet.provider
+}
+
+export function getWalletAddress(): string | null {
+  return CurrentConfig.env === Environment.WALLET_EXTENSION
+    ? walletExtensionAddress
+    : wallet.address
+}
+
+export async function sendTransaction(
+  transaction: ethers.providers.TransactionRequest
+): Promise<TransactionState> {
+  if (CurrentConfig.env === Environment.WALLET_EXTENSION) {
+    return sendTransactionViaExtension(transaction)
+  } else {
+    transaction.value = BigNumber.from(transaction.value)
```

Comment:
Should we check for `null` here?

Reply:
Got the wrong template file, will be fixed by using the one in minting!